schemas/dns.yml

```diff
@@ -0,0 +1,254 @@
+        the domain name used in the question.
+
+    - name: flags.recursion_available
+      level: extended
+      type: boolean
+      description: >
+        A DNS flag specifying whether recursive query support is available in the
+        name server.
+
+    - name: flags.recursion_desired
+      level: extended
+      type: boolean
+      description: >
+        A DNS flag specifying that the client directs the server to pursue a
+        query recursively. Recursive query support is optional.
+
+    - name: flags.authentic_data
+      level: extended
+      type: boolean
+      description: >
+        A DNS flag specifying that the recursive server considers the response
+        authentic.
+
+    - name: flags.checking_disabled
+      level: extended
+      type: boolean
+      description: >
+        A DNS flag specifying that the client disables the server
+        signature validation of the query.
+
+    - name: flags.truncated_response
+      level: extended
+      type: boolean
+      description: >
+        A DNS flag specifying that only the first 512 bytes of the reply were
+        returned.
+
+    - name: response_code
+      level: extended
+      type: keyword
+      description: The DNS status code.
+      example: NOERROR
+
+    - name: question.name
+      level: core
+      type: keyword
+      description: >
+        The domain name being queried. If the name field contains non-printable
+        characters (below 32 or above 126), then those characters are represented
+        as escaped base 10 integers (\DDD). Back slashes and quotes are escaped.
+        Tabs, carriage returns, and line feeds are converted to \t, \r, and
+        \n respectively.
+      example: www.google.com.
+
+    - name: question.type
+      level: core
+      type: keyword
+      description: The type of records being queried.
+      example: AAAA
+
+    - name: question.class
+      level: core
+      type: keyword
+      description: The class of of records being queried.
+      example: IN
+
+    - name: question.etld_plus_one
+      level: core
+      type: keyword
+      description: The effective top-level domain (eTLD) plus one more label.
+        For example, the eTLD+1 for "foo.bar.golang.org." is "golang.org.".
+        The data for determining the eTLD comes from an embedded copy of the
+        data from http://publicsuffix.org.
+      example: amazon.co.uk.
+
+    - name: question.size
+      level: extended
+      type: keyword
+      description: The size of name being queried (in bytes).
+
+    - name: answers
+      level: extended
+      type: object
+      description: >
+        An array containing a dictionary about each answer section returned by
+        the server.
+
+    - name: answers_count
+      level: extended
+      type: long
+      description: >
+        The number of resource records contained in the `dns.answers` field.
+
+    - name: answers.name
```

Comment:
All fields nested under `dns.answers.*` describe the fields expected in each object of the array of answers.

So I would keep `answers` plural, since it's very often going to be multiple answers. Is that what you meant here?

Reply:
All our fields can contain an array, but we stick to singular. Like `ip` instead of `ips`. How will the user query look like?

Reply:
In our guidelines, we do suggest people try to use proper pluralization for their field names. In this sense (and to avoid an unnecessary break with Packetbeat), I would prefer to keep `dns.answers.*`.

This does highlight that I'm adding `dns.resolved_ip` which is meant to contain multiple IPs. So if we were to adjust capitalization, I'd rather do it on the new field, and name it `dns.resolve_ips`.

Reply:
I think `resolved_ip` follows our convention as it can contain one or multiple entries. Otherwise almost all of our fields would have to be plural. Probably we should take this discussion somewhere else ;-)